oak_idl_tests/tests/oak_abi.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! This crate contains tests for the `oak_idl_gen_structs` and `oak_idl_gen_services` crates. It
+//! needs to be separate from them in order to be able to invoke them at build time.
+
+mod oak {
+    #![allow(clippy::derivable_impls, clippy::needless_borrow)]
+    #![allow(dead_code, unused_imports)]
+
+    include!(concat!(env!("OUT_DIR"), "/oak_generated.rs"));
+    include!(concat!(env!("OUT_DIR"), "/oak_services.rs"));
+}
+
+struct OakImpl;
+
+/// An implementation of the [`Oak`] service trait for testing.
```

Comment:
We are testing the IDL and nothing from Oak Functions, no?

I think it would help me to emphasize that this is a totally independent test. (If that is the case, oc).

Reply:
Seems like an independent test. Agreed that making that clearer would be nice (by swapping out "Oak" in the example for something like `MyService`

Reply:
Renamed to TestService.